### ISSUE TITLE
Kotlin nullability controls don't seem to work for views

### ISSUE THREAD

**Issue by mikehearn** [jOOQ/jOOQ#14928]
### Expected behavior

I tried out the new nullability feature for generated Kotlin data classes and it seems to work for tables, but had no effect for views, where the generated data classes still always had nullable fields. I was expecting there to be no difference.

### Steps to reproduce the problem

I was using PostgreSQL with something like this:

```
CREATE TABLE t_ticket (
  id SERIAL PRIMARY KEY,
  title TEXT NOT NULL,
  description TEXT NOT NULL,
  assignee TEXT
);

-- define the ticket view
CREATE VIEW ticket AS
  SELECT id, title, description, assignee FROM t_ticket;
```

### jOOQ Version

3.18

### Database product and version

PostgreSQL 14

### Java Version

Java 19/20 (both)

### OS Version

macOS 13

### JDBC driver name and version (include name if unofficial driver)

pgjdbc-ng

**Comment by lukaseder** [comment]
Thanks for your report. Try this query, which is equivalent to what jOOQ's code generator does behind the scenes:

```sql
SELECT column_name, is_nullable
FROM information_schema.columns
WHERE table_name = 'ticket'
ORDER BY ordinal_position;
```

It produces:

```
|column_name|is_nullable|
|-----------|-----------|
|id         |YES        |
|title      |YES        |
|description|YES        |
|assignee   |YES        |
```

Alternatively:

```sql
SELECT a.attname, a.attnotnull
FROM pg_class AS c
JOIN pg_attribute AS a ON a.attrelid = c.oid
WHERE c.relname = 'ticket'
ORDER BY a.attnum;
```

Producing:

```
|attname    |attnotnull|
|-----------|----------|
|id         |false     |
|title      |false     |
|description|false     |
|assignee   |false     |
```

So, if PostgreSQL can't tell us anything about view column nullability, then I don't think jOOQ can do much about it, currently.

There's a set of feature requests where we analyse this type of nullability (and other type information) propagation through view definitions ourselves, see e.g.:

- https://github.com/jOOQ/jOOQ/issues/11070
- https://github.com/jOOQ/jOOQ/issues/11081

It would then be possible to reverse engineer the *actual* row type of a stored view based on its source code, if available, but we're currently not doing that.

**Comment by lukaseder** [comment]
@MuhamedMagdi Have you read the issue comments and the conclusion?

**Comment by MuhamedMagdi** [comment]
Yes, I've read through it. However, I wanted to confirm specifically if the feature mentioned in the quoted message has been implemented since it wasn't explicitly addressed. The quote I'm referring to is: 
> It would then be possible to reverse engineer the actual row type of a stored view based on its source code, if available, but we're currently not doing that.

Thank you for your assistance.

**Comment by lukaseder** [comment]
I see. Nothing has been done in this area. I've created a new issue to track this:

- https://github.com/jOOQ/jOOQ/issues/16528

No promises with respect to prioritisation. It's a big task.
